app/presenters/document_export_presenter.rb

```diff
@@ -46,6 +49,14 @@ def complete_images_hash(edition_images)
     end
   end
 
+  def complete_attachments_hash(edition_attachments)
+    edition_attachments.map do |attachment|
+      if attachment.is_a?(FileAttachment)
+        attachment.as_json(include: :attachment_data, methods: :url)
+      end
```

Comment:
This is going to leave nil values in the array of attachments if items are not file attachments. 

When I suggested nuke I was thinking more like an exception but that feels a bit fishy given we're now unlikely to have the next part of the work done anytime soon, and it shouldn't be too much work to get it working for all 3 attachments.

I think we can get this resolved with:

```
edition_attachments.map do |attachment|
  attachment.as_json(include: :attachment_data, methods: %i[url type]).tap do |json|
     json.merge!(govspeak_content: attachment.govspeak_content.as_json) if attachment.respond_to?(:govspeak_content)
  end
end
```

The type in there is so that it returns "FileAttachment", "ExternalAttachment", or "HtmlAttachment" - govspeak content is an association only HtmlAttachments has.

and then we'd just need tests that cover each attachment type.

Reply:
@Tetrino I see you marked this as resolved, but you may have missed the bit about adding tests for other attachment types.

Reply:
@kevindew oops, you're right I apologise. Head's not been screwed on.